### PR TITLE
lprobe: init at 0.1.3

### PR DESCRIPTION
**Pull request by cageyv**
https://github.com/fivexl/lprobe/releases/tag/v0.1.3

## Description of changes

- LProbe: A command-line tool to perform Local Health Check Probes inside Container Images (ECS, Docker)
- [Blog post with more info LProbe: A One-Stop-Shop Tool for Secure, Local Health Checks](https://fivexl.io/blog/lprobe/ )

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---
Screenshot 2024-08-25 at 13 25 11

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by cageyv** [comment]
@ofborg build

**Comment by cageyv** [comment]
cc @wineee @vbgl @matteo-pacini
It works as expected. Screenshot attached. 

Long story short. Docker Desktop Debug uses NixOS. Lprobe is my default tool for Health Checks inside containers.

File: maintainers/team-list.nix [hunk 1]
```diff
@@ -293,6 +293,16 @@ with lib.maintainers;
...
+  fivexl = {
+    members = [
+      cageyv
+    ];
+    scope = "Maintain lprobe";
+    shortName = "fivexl";
+    enableFeatureFreezePing = false;
+    githubTeams = [ "fivexl" ];
+  };
+
```
Comment (matteo-pacini) [inline maintainers/team-list.nix:305]: Do we need a team to maintain a single package?

I would expect a team for things like GNOME, GPU drivers, etc.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -3177,6 +3177,15 @@
...
   };
+  cageyv = {
+    email = "pingme@cageyv.dev";
+    github = "cageyv";
+    githubId = 51059484;
+    name = "Vladmir Samoylov";
+    keys = [
+      { fingerprint = "8916 F727 734E 77AB 437F  A33A 19AB 76F5 CEE1 1392"; }
+    ];
+  };
```
Comment (matteo-pacini) [inline maintainers/maintainer-list.nix:3188]: This needs to be in a separate commit as per Nixpkgs convention, i.e. `maintainers: add cageyv`

File: pkgs/tools/networking/lprobe/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  ];
+
+  meta = with lib; {
+    description = "A command-line tool to perform Local Health Check Probes inside Container Images (ECS, Docker)";
+    homepage = "https://github.com/fivexl/lprobe";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ cageyv ];
+    mainProgram = "lprobe";
+  };
+}
```
Comment (matteo-pacini) [inline pkgs/tools/networking/lprobe/default.nix:32]: This file is a duplicate, please remove it - `pkgs/by-name` should be preferred when creating new packages

File: pkgs/by-name/lp/lprobe/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+buildGoModule rec {
+  pname = "lprobe";
+  version = "0.1.3";
+
+  src = fetchFromGitHub {
+    owner = "fivexl";
+    repo = "lprobe";
+    rev = "refs/tags/v${version}";
+    sha256 = "sha256-q7lH0aLgQNM4jrrrq2ua+pt4/VknxlzKzDH5J4MwjfA=";
```
Comment (matteo-pacini) [inline pkgs/by-name/lp/lprobe/package.nix:16]:
```suggestion
    hash = "sha256-q7lH0aLgQNM4jrrrq2ua+pt4/VknxlzKzDH5J4MwjfA=";
```

File: pkgs/by-name/lp/lprobe/package.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    libpcap
+  ];
+
+  meta = with lib; {
+    description = "A command-line tool to perform Local Health Check Probes inside Container Images (ECS, Docker)";
+    homepage = "https://github.com/fivexl/lprobe";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ cageyv ];
+    mainProgram = "lprobe";
+  };
```
Comment (matteo-pacini) [inline pkgs/by-name/lp/lprobe/package.nix:31]:
```suggestion
  meta = {
    description = "A command-line tool to perform Local Health Check Probes inside Container Images (ECS, Docker)";
    homepage = "https://github.com/fivexl/lprobe";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ cageyv ];
    mainProgram = "lprobe";
  };
```
As suggested in: https://github.com/NixOS/nixpkgs/issues/208242

`platforms` is missing also, that should be added.

**Review by matteo-pacini — CHANGES_REQUESTED** [review]
Thank you for your first contribution,  I just left a few comments.

Happy to help / elaborate further if requested changes are not clear 🙏🏻

**Comment by cageyv** [comment]
@matteo-pacini 

Thank you for your suggestions and guidelines. 🙏🏻

- team removed
- maintainer in separate commit 
- duplicate removed
- platforms added
- hash fixed

**Review by matteo-pacini — APPROVED** [review]
LGTM, thanks for addressing my comments

**Review by wineee — APPROVED** [review]

**Pull request merged**